src/lib/proof-system/zkprogram.ts

```diff
@@ -787,14 +919,43 @@ function picklesRuleFromFunction(
       let input = fromFieldVars(publicInputType, publicInput);
       result = await func(input, ...finalArgs);
     }
+
+    proofs.forEach(({ proofInstance, classReference }, index) => {
+      if (proofInstance instanceof DynamicProof) {
```

Comment:
nit: early return if `!(proofInstance instanceof DynamicProof)` would be cleaner